app/services/oidc_jwt_validate.rb

```diff
@@ -12,8 +12,8 @@ def decoded_payload
         verify_aud: true,
         iss: Setting['oidc_issuer'],
         verify_iss: true,
-        algorithms: [Setting['oidc_algorithm']],
-        jwks: jwks_loader }
+        algorithms: [Setting['oidc_algorithm']] },
+      &method(:jwk_findkey)
```

Comment:
I don't like the instance method exposure, it is not technically wrong, but it is quite a code smell.
Can't we have an anonymos method instead? The anonymous method can be build in the instance method:
```ruby
  def jwk_findkey
    finder = JWT::JWK::KeyFinder.new(jwks: jwks_loader)
    # Support ADFS's use of the non-standard x5t header for holding key ID
    ->(header) {
      key = finder.key_for(header['kid']) if header.key? 'kid'
      key ||= finder.key_for(header['x5t']) if header.key? 'x5t'
      key
    }
  end
```